backstory/src/test/java/com/google/sps/servlets/StoryManagerTest.java

```diff
@@ -0,0 +1,150 @@
+  }
+
+  @Test
+  public void invalidPrefix() {
+    // Verifies that "" replaces invalid prefix input.
+    storyManager = new StoryManagerImpl(null, SIZE_SAMPLE, TEMPERATURE_SAMPLE);
+
+    String expected = "";
+    String actual = storyManager.getPrefix();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void lengthTooLow() {
+    // Verifies that 100 replaces invalid low size input.
+    storyManager = new StoryManagerImpl(PREFIX_SAMPLE, 0, TEMPERATURE_SAMPLE);
+
+    int expected = 100;
+    int actual = storyManager.getMaxLength();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void lengthTooHigh() {
+    // Verifies that 1000 replaces invalid low size input.
+    storyManager = new StoryManagerImpl(PREFIX_SAMPLE, 10000, TEMPERATURE_SAMPLE);
+
+    int expected = 1000;
+    int actual = storyManager.getMaxLength();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void invalidTemperature() {
+    // Verifies that 1 replaces invalid temperature input
+    storyManager = new StoryManagerImpl(PREFIX_SAMPLE, SIZE_SAMPLE, 1000.0);
+
+    Double expected = 1.0;
+    Double actual = storyManager.getTemperature();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void correctRequestConfiguration() throws IOException {
+    /* Ensure the method configures a proper POST request and receives
+     * proper output translation.
+     */
+
+    // Declare StoryManager with standard inputs.
+    storyManager = new StoryManagerImpl(PREFIX_SAMPLE, SIZE_SAMPLE, TEMPERATURE_SAMPLE);
+
+    // Configure fake Request Factory to allow for mock request injection.
+    StoryManagerRequestFactoryFakeImpl factoryFake = new StoryManagerRequestFactoryFakeImpl();
+    HttpRequest mockRequest = mock(HttpRequest.class);
+    factoryFake.setRequest(mockRequest);
+    storyManager.setRequestFactory(factoryFake);
+
+    // Configure mockHeaders to ensure correct type setting of request headers.
+    HttpHeaders mockHeaders = mock(HttpHeaders.class);
+    when(mockRequest.getHeaders()).thenReturn(mockHeaders);
+
+    // Define expected input JSON for post request.
+    String expectedRequestString = "{\"length\": " + SIZE_SAMPLE
+        + ",\"truncate\": \"<|endoftext|>\", \"prefix\": \"" + PREFIX_SAMPLE
+        + "\", \"temperature\": " + TEMPERATURE_SAMPLE + "}";
+
+    // Stub the mockResponse to return specified text output.
+    String sampleJSONOutput = "{\"text\": \"foo\"}";
+    HttpResponse mockResponse = mock(HttpResponse.class);
+    try {
+      when(mockRequest.execute()).thenReturn(mockResponse);
+      when(mockResponse.parseAsString()).thenReturn(sampleJSONOutput);
+    } catch (IOException mockHttpException) {
+      throw new RuntimeException("Error with mock Http", mockHttpException);
+    }
+
+    // Define expected output text for generateText()
+    String expectedOutput = "foo";
```

Comment:
I'd suggest declaring this above, and using it to construct `sampleJSONOutput` to clarify the link between the two:

```
String expectedOutput = "foo";
String sampleJSONOutput = "{\"text\": \"" + expectedOutput + "\"}";
```

Reply:
Done